specification.md

```diff
@@ -4,73 +4,73 @@ Opportunity:
 * Any number of auxiliary **dataframes** and **arrays**. These can contain anything (e.g., `reducedDims` in SingleCellExperiment) but should be able to efficiently join with `obs` and `var` to produce the labels to query `X`.
 
-The following image summarizes the model for an **sc_group**, in the case where a set of `X` array layers shares the `obs` and `var` dataframes. This can be generalized to _multiple different groups_ of what is depicted below to capture cases where the `X` array layers do not share the `obs` and `var` dataframes.
+The following image summarizes the model for a **SOMA**, in the case where a set of `X` array layers shares the `obs` and `var` dataframes. This can be generalized to _multiple different groups_ of what is depicted below to capture cases where the `X` array layers do not share the `obs` and `var` dataframes.
 
 ![matrix schema](./images/matrix.png)
 
-And for the entire **sc_dataset**, containing one or more named **sc_group**:
+And for the entire **SOMA collection**, containing one or more named **SOMAs**:
 
 ```
-sc_dataset                     # sc_dataset
-    |_ sc_group_1              # sc_group (contents as in the figure)
-    |_ sc_group_2              # sc_group (contents as in the figure)
+soma_collection            # SOMACollection
+    |_ soma_1              # SOMA (contents as in the figure)
+    |_ soma_2              # SOMA (contents as in the figure)
     |_ ...
 ```
 
 In the most basic case, we need three queries:
 
-* The first uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `obs` dataframe and produces a set of **row labels**, let’s call them  `obs_labels`.
-* The second uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `var` dataframe and produces a set of **column labels**, let’s call them  `var_labels`.
+* The first uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `obs` dataframe and produces a set of **row labels** -- let’s call them  `obs_labels`.
+* The second uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `var` dataframe and produces a set of **column labels** -- let’s call them  `var_labels`.
 * Then the query will take as input an `X` array and potentially one of its “layers” (let’s denote that by `X[layer]`), `obs_labels` and `var_labels`.
 * The result of this query `Q(X[layer], obs_labels, var_labels)` can be represented in two forms:
     * **Dense**, e.g., a 2D numpy array
     * **Sparse**, e.g.,
-        * 3 1D numpy arrays, one for the row indices, one of the column indices, and one for the `X` values, or
+        * Three 1-D numpy arrays, one for the row indices, one of the column indices, and one for the `X` values, or
         * A pandas dataframe indexed on row and column indices for fast search
 
 ## Exemplary Use Cases
```

Comment:
I'd delete code samples and reference working notebooks. It's too easy for code snippets to either not work or stop working and readers tend to copy and paste and then become frustrated when code fails.